editor_canvas.py

```
@@ -82,53 +148,137 @@ def find_class_at(self, x, y) -> UMLClass | None:
+                # Calculate distance from cursor center to intersection point
+                dist = geometry.distance((x, y), rect_hit)
+                hits.append((dist, rel))
+        
+        if hits:
+            # Return the closest one
+            hits.sort(key=lambda item: item[0])
+            return hits[0][1]
+            
+        return None
+
     def on_button_press(self, event):
```

Ruff (0.15.6)

[warning] Too many branches (14 > 12)

(PLR0912)